test/images/netproxy/netproxy.go

```diff
@@ -0,0 +1,228 @@
+		udpAddress, err = net.ResolveUDPAddr("udp", hostPort)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("host and/or port param are invalid. %v", err), http.StatusBadRequest)
+			return
+		}
+	}
+	errors := make([]string, 0)
+	responses := make([]string, 0)
+	var response string
+	for i := 0; i < tries; i++ {
+		if protocol == "udp" {
+			response, err = dialUDP(request, udpAddress)
+		} else {
+			response, err = dialHTTP(request, hostPort)
+		}
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("%v", err))
+		} else {
+			responses = append(responses, response)
+		}
+	}
+	output := map[string][]string{}
+	if len(response) > 0 {
+		output["responses"] = responses
+	}
+	if len(errors) > 0 {
+		output["errors"] = errors
+	}
+	bytes, err := json.Marshal(output)
+	if err == nil {
+		fmt.Fprintf(w, string(bytes))
+	} else {
+		http.Error(w, fmt.Sprintf("response could not be serialized. %v", err), http.StatusExpectationFailed)
+	}
+}
+
+func dialHTTP(request, hostPort string) (string, error) {
+	transport := &http.Transport{}
+	httpClient := createHTTPClient(transport)
+	resp, err := httpClient.Get(fmt.Sprintf("http://%s/%s", hostPort, request))
+	defer transport.CloseIdleConnections()
+	if err == nil {
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err == nil {
+			return string(body), nil
+		}
+	}
+	return "", err
+}
+
+func createHTTPClient(transport *http.Transport) *http.Client {
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   5 * time.Second,
+	}
+	return client
+}
+
+func dialUDP(request string, remoteAddress *net.UDPAddr) (string, error) {
+	Conn, err := net.DialUDP("udp", nil, remoteAddress)
+	if err != nil {
+		return "", fmt.Errorf("udp dial failed. err:%v", err)
+	}
+
+	defer Conn.Close()
+	buf := []byte(request)
+	_, err = Conn.Write(buf)
+	if err != nil {
+		return "", fmt.Errorf("udp connection write failed. err:%v", err)
+	}
+	udpResponse := make([]byte, 1024)
+	Conn.SetReadDeadline(time.Now().Add(5 * time.Second))
+	count, err := Conn.Read(udpResponse)
+	if err != nil || count == 0 {
+		return "", fmt.Errorf("reading from udp connection failed. err:'%v'", err)
+	}
+	return string(udpResponse[0:count]), nil
+}
+
+func shellHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println(r.FormValue("shellCommand"))
+	output, err := exec.Command(shellPath, "-c", r.FormValue("shellCommand")).CombinedOutput()
+	assertNoError(err)
+	fmt.Fprintf(w, string(output))
+}
+
+func hostNameHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, getHostName())
+}
+
+func startUDPServer(udpPort int) {
+	serverAddress, err := net.ResolveUDPAddr("udp", fmt.Sprintf(":%d", udpPort))
+	assertNoError(err)
+	serverConn, err := net.ListenUDP("udp", serverAddress)
+	defer serverConn.Close()
+	buf := make([]byte, 1024)
+
+	for {
+		n, clientAddress, err := serverConn.ReadFromUDP(buf)
+		assertNoError(err)
+		receivedText := strings.TrimSpace(string(buf[0:n]))
+		if receivedText == "hostName" {
```

Comment:
so UDP only supports hostname but HTTP also supports "shell" and "dial"?  Just being clear - maybe a comment?